SeaPublicWebsite.UnitTests/EpcTests.cs

```diff
@@ -22,1346 +22,411 @@ public void CanParseEpcDto(EpcTestCase testCase)
-                    "flat"
-                }
-            },
-            new Epc
-            {
-                RoofConstruction = RoofConstruction.Flat
-            }),
-        new(
-            "Can parse roof construction mixed",
-            new EpbEpcAssessmentDto
-            {
-                AssessmentType = "RdSAP",
-                RoofDescription = new List<string>
-                {
-                    "pitched",
-                    "flat"
-                }
-            },
-            new Epc
-            {
-                RoofConstruction = RoofConstruction.Mixed
-            }),
-    };
+                    RoofConstruction = p.outputRoofConstruction
+                }))
+            .ToArray();
     
-    private static readonly EpcTestCase[] RoofInsulationTestCases =
-    {
-        new(
-            "Can handle null roof insulation",
-            new EpbEpcAssessmentDto
-            {
-                AssessmentType = "RdSAP",
-                RoofDescription = null
-            },
-            new Epc
-            {
-                RoofInsulated = null
-            }),
-        new(
-            "Can parse roof insulated",
-            new EpbEpcAssessmentDto
-            {
-                AssessmentType = "RdSAP",
-                RoofDescription = new List<string>
-                {
-                    "flat, insulated"
-                }
-            },
-            new Epc
-            {
-                RoofConstruction = RoofConstruction.Flat,
-                RoofInsulated = RoofInsulated.Yes
-            }),
-        new(
-            "Can parse roof insulated >= 200mm",
-            new EpbEpcAssessmentDto
-            {
-                AssessmentType = "RdSAP",
-                RoofDescription = new List<string>
-                {
-                    "flat, 200 mm loft insulation"
-                }
-            },
-            new Epc
-            {
-                RoofConstruction = RoofConstruction.Flat,
-                RoofInsulated = RoofInsulated.Yes
-            }),
-        new(
-            "Can parse roof uninsulated",
-            new EpbEpcAssessmentDto
-            {
-                AssessmentType = "RdSAP",
-                RoofDescription = new List<string>
+    private static readonly EpcTestCase[] RoofInsulatedTestCases =
+        new (string Descrption, string[] inputRoofDescription, RoofInsulated? outputRoofInsulated)[]
+            {
+                ("Can parse roof insulated", new[] {"pitched, insulated"}, RoofInsulated.Yes),
+                ("Can parse roof insulated (limited)", new[] {"pitched, limited insulation (assumed)"}, RoofInsulated.Yes),
+                ("Can parse roof insulated >= 200mm", new[] {"pitched, 200 mm loft insulation"}, RoofInsulated.Yes),
+                ("Can parse roof uninsulated", new[] {"pitched, no insulation"}, RoofInsulated.No),
+                ("Can parse roof uninsulated < 200mm", new[] {"pitched, 150 mm loft insulation"}, RoofInsulated.No),
```

Comment:
This seems to be missing quite a few of the spreadsheet rows

Reply:
We don't care if a roof is flat about its insulation (we don't ask the user in that case). So we only need to test the pitched ones (I've added 'insulated at rafters')